File: smarttrade_app/lib/features/chart_analysis/data/repositories/macro_news_repository_impl.dart
import 'package:dartz/dartz.dart';
import 'package:smarttrade_app/core/error/failures.dart';
import 'package:smarttrade_app/features/chart_analysis/domain/entities/macro_news_headline.dart';
import 'package:smarttrade_app/features/chart_analysis/domain/repositories/macro_news_repository.dart';

/// Placeholder implementation: returns a fixed sample so the ticker has
/// something to scroll in the MVP. Replace with a data source that hits
/// a real macroeconomic calendar/news API once one is chosen.
class MacroNewsRepositoryImpl implements MacroNewsRepository {
  const MacroNewsRepositoryImpl();

  @override
  Future<Either<Failure, List<MacroNewsHeadline>>> getLatestHeadlines() async {
    final now = DateTime.now();
    return Right([
      MacroNewsHeadline(
        headline: 'FOMC mantém juros; mercado aguarda fala de Powell.',
        impact: MacroImpact.high,
        publishedAt: now,
      ),
      MacroNewsHeadline(
        headline: 'CPI dos EUA em linha com o consenso.',
        impact: MacroImpact.medium,
        publishedAt: now,
      ),
      MacroNewsHeadline(
        headline: 'Payroll acima do esperado pressiona yields.',
        impact: MacroImpact.high,
        publishedAt: now,
      ),
      MacroNewsHeadline(
        headline: 'Feed de dados macro em desenvolvimento.',
        impact: MacroImpact.low,
        publishedAt: now,
      ),
    ]);
  }
}

File: smarttrade-web/src/features/chart-analysis/data/repositories/macroNewsRepositoryImpl.ts
import { ok, type Result } from '@/core/result/result'
import type { MacroNewsHeadline } from '../../domain/entities/macroNewsHeadline'
import type { MacroNewsRepository } from '../../domain/repositories/macroNewsRepository'

/**
 * Placeholder implementation: returns a fixed sample so the ticker has
 * something to scroll in the MVP. Replace with a data source that hits
 * a real macroeconomic calendar/news API once one is chosen.
 */
export class MacroNewsRepositoryImpl implements MacroNewsRepository {
  async getLatestHeadlines(): Promise<Result<MacroNewsHeadline[]>> {
    const now = new Date()
    return ok([
      {
        headline: 'FOMC mantém juros; mercado aguarda fala de Powell.',
        impact: 'high',
        publishedAt: now,
      },
      {
        headline: 'CPI dos EUA em linha com o consenso.',
        impact: 'medium',
        publishedAt: now,
      },
      {
        headline: 'Payroll acima do esperado pressiona yields.',
        impact: 'high',
        publishedAt: now,
      },
      {
        headline: 'Feed de dados macro em desenvolvimento.',
        impact: 'low',
        publishedAt: now,
      },
    ])
  }
}

export const macroNewsRepository: MacroNewsRepository = new MacroNewsRepositoryImpl()
